src/components/internals/root-ref/root-ref.js

```diff
@@ -0,0 +1,48 @@
+// inspired from https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/RootRef/RootRef.js
```

Comment:
Can we also copy over the comments, just in case they get deleted from the source? Then we have a bit more context about the scope of this component. Thanks